File: kotlin/src/main/kotlin/web/client/HttpClientInterface.kt
package web.client

import io.ktor.http.*
import web.snapshots.Snapshot
import web.url.Url

interface HttpClientInterface {
    fun fetch(
        url: Url,
        method: HttpMethod = HttpMethod.Get,
        addHeaders: Map<String, String> = mapOf(),
        payload: String? = null,
    ): Snapshot

    fun getSingleCookieValue(url: String, cookieName: String): String?
}

File: symfony/src/Utils/Web/HttpClient/HttpClientInterface.php
<?php

declare(strict_types=1);

namespace App\Utils\Web\HttpClient;

use App\Utils\Web\Snapshots\Snapshot;
use App\Utils\Web\Url;
use Veelkoov\Debris\Maps\StringToString;

interface HttpClientInterface
{
    public function fetch(
        Url $url,
        string $method = 'GET',
        StringToString $addHeaders = new StringToString(),
        ?string $content = null,
    ): Snapshot;

    public function getSingleCookieValue(string $cookieName, string $domain): ?string;
}
